marketplace/app/v0/object_storage.py

```diff
@@ -0,0 +1,206 @@
+import json
+
+import marketplace_standard_app_api.models.object_storage as ObjectStorageModel
+from fastapi import Request, UploadFile
+
+from ..utils import check_capability_availability
+from .base import _MarketPlaceAppBase
+
+
+class MarketPlaceObjectStorageApp(_MarketPlaceAppBase):
+    @check_capability_availability
+    def list_collections(
+        self, limit: int = 100, offset: int = 0
+    ) -> ObjectStorageModel.CollectionListResponse:
+        params = {"limit": limit, "offset": offset}
+        return ObjectStorageModel.CollectionListResponse(
+            **self._client.get("listCollections", params).json()
+        )
+
+    @check_capability_availability
+    def list_datasets(
+        self,
+        collection_name: ObjectStorageModel.CollectionName,
+        limit: int = 100,
+        offset: int = 0,
+    ) -> ObjectStorageModel.DatasetListResponse:
+        params = {"collection_name": collection_name, "limit": limit, "offset": offset}
+        return ObjectStorageModel.DatasetListResponse(
+            **self._client.get("listDatasets", params=params).json()
+        )
+
+    # TODO: check this implementation
+    @check_capability_availability
+    def create_or_update_collection(
+        self,
+        request: Request,
+        collection_name: ObjectStorageModel.CollectionName = None,
+    ):
+        params = {}
+        if collection_name:
+            params = {"collection_name": collection_name}
+        return self._client.put(
+            "createOrUpdateCollection",
+            params=params,
+            headers=request.headers,
+            json=request.json(),
+        ).json()
+
+    @check_capability_availability
+    def delete_collection(self, collection_name: ObjectStorageModel.CollectionName):
+        params = {"collection_name": collection_name}
+        return self._client.delete("deleteCollection", params=params).json()
+
+    # TODO: check implementation
+    @check_capability_availability
+    def get_collection_metadata(
+        self, collection_name: ObjectStorageModel.CollectionName
+    ):
+        params = {"collection_name": collection_name}
+        response_headers: dict = self._client.get(
+            "getCollectionMetadata", params=params
+        ).headers
+        return json.dumps(
+            dict(
+                filter(
+                    lambda val: str(val[0]).startswith("X-Object-Meta"),
+                    response_headers.items(),
+                )
+            )
+        )
```

Comment:
I think a dict-comprehension would be slightly more obvious to understand and I think there are some minor issues with the current implementation:
```suggestion
        return {
            re.sub(r"^X-Object-Meta-", "", name): json.loads(value)
            for key, value in response_headers.items()
            if key.startswith("X-Object-Meta-")
        }

```
As suggested before, this could be placed in a `_decode_metadata_headers(headers: dict) -> dict` utility function.